include/crm/fencing/internal.h

```diff
@@ -134,6 +138,7 @@ xmlNode *create_device_registration_xml(const char *id, const char *namespace, c
 #  define stonith_channel            "st_command"
 #  define stonith_channel_callback   "st_callback"
 
-#  define STONITH_WATCHDOG_AGENT  "#watchdog"
+#  define STONITH_WATCHDOG_AGENT  "fence_watchdog"
+#  define STONITH_WATCHDOG_ID     "watchdog"
```

Comment:
Now that we're allowing explicit configuration, do we need a constant ID? I worry that things will only work at the moment if someone names their fence_watchdog device "watchdog" and nothing else.

Reply:
Yes it will just work if it is called "watchdog" and I think that isn't a bad idea - anyway I'm enforcing that when building up the device-list. Could we already catch something like that in the schema?

Reply:
Schema is probably not the best place, since technically someone can register via API as well. Maybe if someone registers a device with fence_watchdog under some other name, we could log a config error and map it to "watchdog" (and also config error if someone registers "watchdog" with any other device). Either that, or track the name they give it. In any case, it would probably also be good to give a config error if there is more than one fence_watchdog device registered.

Reply:
Didn't mean schema as a replacement for catching it later as well. Just to catch it as early as possible and to give the tools a chance to moan right away.
Like forcing it to be named "watchdog" there is no way to have more than one - which as well speaks against the mapping as this would open up the possibility to have more of them configured that are folded into one with unexpected results.
As written the check and error-log are already there.